src/Component.js

```diff
@@ -1,274 +1,122 @@
 import * as Utility from './Utility.js'
 import * as Element from './Element.js'
-import * as Assert from './Assert.js'
-import * as Lifecycle from './Lifecycle.js'
 import * as Exception from './Exception.js'
-import * as Node from './Node.js'
 import * as Reconcile from './Reconcile.js'
 import * as Schedule from './Schedule.js'
-import * as Registry from './Registry.js'
 
 /**
- * @type {object}
+ * @param {object} prev
+ * @param {object} next
+ * @return {boolean}
  */
-export var descriptors = {
-	/**
-	 * @param {object} state
-	 * @param {function} callback
-	 * @return {object}
-	 */
-	setState: {
-		value: function (state, callback) {
-			return dispatch(Registry.get(this), this, state, callback)
+export function compare (prev, next) {
+	if (prev !== next) {
+		for (var key in prev) {
+			if (!Utility.has(next, key)) {
+				return false
+			}
 		}
-	},
-	/**
-	 * @param {function} callback
-	 * @return {object}
-	 */
-	forceUpdate: {
-		value: function (callback) {
-			return this.setState(Enum.obj, callback)
+		for (var key in next) {
+			if (!Utility.is(prev[key], next[key])) {
+				return false
+			}
 		}
 	}
-}
-
-/**
- * @constructor
- */
-export var struct = Utility.extend(factory(), Utility.assign({
-	/**
-	 * @type {function}
-	 */
-	render: {value: render}
-}, descriptors))
-
-/**
- * @constructor
- */
-export var pure = Utility.extend(factory(), Utility.assign({
-	/**
-	 * @type {function}
-	 */
-	render: {value: render},
-	/**
-	 * @type {function}
-	 */
-	shouldComponentUpdate: {value: compare}
-}, descriptors))
 
-/**
- * @param {object} props
- * @param {object} state
- * @param {object} context
- * @return {object?}
- */
-export function render (props, state, context) {
-	return typeof props.children === 'function' ? props.children(props, state, context) : props.children
-}
-
-/**
- * @param {object} props
- * @param {object} state
- * @param {object} context
- * @return {boolean}
- */
-export function compare (props, state, context) {
-	return Utility.compare(this.props, props) || Utility.compare(this.state, state) || Utility.compare(this.context, context)
+	return true
 }
 
 /**
+ * @param {function} children
+ * @param {function} callback
  * @return {function}
  */
-export function factory () {
-	return function constructor (props) {
-		this.props = props
-		this.state = null
-		this.context = null
-	}
+export function memo (children, callback) {
+	return memoize(children, callback || compare)
 }
 
 /**
- * @param {function} value
- * @param {object?} proto
+ * @param {function} children
+ * @param {function} callback
  * @return {function}
  */
-export function identity (value, proto) {
-	if (proto) {
-		if (proto.setState) {
-			return value
-		} else if (proto.render) {
-			return Utility.extend(value, descriptors)
+export function memoize (children, callback) {
+	return function (props, state) {
+		if (this.parent !== null && this.value === null && callback(this.props, props) === true) {
```

Comment:
why `this.parent !== null`?

Reply:
The branch is only executed on updates, `this.parent` is never null after the initial mount.